ios/sdk/src/JitsiMeetManager/PiPWindow/PiPWindow.swift

```diff
@@ -0,0 +1,161 @@
+    }
+    
+    /// Resize the root view to full screen
+    open func goToFullScreen() {
+        isInPiP = false
+        animateRootViewChange()
+        dragController.stopDragListener()
+        
+        // hide PiP UI
+        exitPiPButton?.removeFromSuperview()
+        exitPiPButton = nil
+        
+        // remove gesture
+        tapGestureRecognizer?.removeTarget(self, action: #selector(toggleExitPiP))
+        tapGestureRecognizer = nil
+    }
+    
+    /// Stop the dragging gesture of the root view
+    public func stopDragGesture() {
+        dragController.stopDragListener()
+    }
+    
+    /// Customize the presentation of exit pip button
+    open func configureExitPiPButton(target: Any, action: Selector) -> UIButton {
+        let buttonImage = UIImage.init(named: "image-resize", in: Bundle(for: type(of: self)), compatibleWith: nil)
+        let button = UIButton(type: .custom)
+        let size: CGSize = CGSize(width: 44, height: 44)
+        button.setImage(buttonImage, for: .normal)
+        button.backgroundColor = .gray
+        button.layer.cornerRadius = size.width / 2
+        button.frame = CGRect(origin: CGPoint.zero, size: size)
+        if let view = rootViewController?.view {
+            button.center = view.convert(view.center, from:view.superview)
+        }
+        button.addTarget(target, action: action, for: .touchUpInside)
+        return button
+    }
+    
+    // MARK: - Manage presentation switching
+    
+    private func animateRootViewChange() {
+        UIView.animate(withDuration: 0.25) {
+            self.rootViewController?.view.frame = self.changeRootViewRect()
+            self.rootViewController?.view.setNeedsLayout()
+        }
+    }
+    
+    private func changeRootViewRect() -> CGRect {
+        guard isInPiP else {
+            return self.bounds
+        }
+        
+        // resize to suggested ratio and position to the bottom right
+        let adjustedBounds = UIEdgeInsetsInsetRect(self.bounds, dragBoundInsets)
+        let size = CGSize(width: bounds.size.width * pipSizeRatio,
+                          height: bounds.size.height * pipSizeRatio)
+        let x: CGFloat = adjustedBounds.maxX - size.width
+        let y: CGFloat = adjustedBounds.maxY - size.height
+        return CGRect(x: x, y: y, width: size.width, height: size.height)
+    }
+    
+    // MARK: - Exit PiP
+    
+    @objc private func toggleExitPiP() {
+        guard let view = rootViewController?.view else { return }
+        
+        if exitPiPButton == nil {
+            // show button
+            let button = configureExitPiPButton(target: self, action: #selector(exitPiP))
+            view.addSubview(button)
+            exitPiPButton = button
+            
+        } else {
+            // hide button
+            exitPiPButton?.removeFromSuperview()
+            exitPiPButton = nil
+        }
+        
+    }
+    
+    @objc private func exitPiP() {
+        goToFullScreen()
```

Comment:
Why are the two names `exitPiP` and `goToFullScreen` for what seems to be one and the same abstraction necessary? We already have `enterPictureInPicture` as a name introduced on the JavaScript and Objective-C sides. Then I'd expect `goToPiP` to actually be called `enterPictureInPicture`. The opposite of `enter` is usually exit so I'd expect to see `exitPictureInPicture` not both `exitPiP` and `goTOFullScreen`.

Reply:
I will rename the function `goToFullScreen()` function to be `exitPictureInPicture()` and just use this function in the `exit pip button` itself instead of proxying that into another function.